eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -0,0 +1,300 @@
+     * If the connection string is copied from the Event Hubs namespace, it will likely not contain the path to the
+     * desired Event Hub, which is needed. In this case, the path can be added manually by adding
+     * {@literal "EntityPath=EVENT_HUB_NAME"} to the end of the connection string. For example,
+     * "EntityPath=telemetry-hub".
+     * </p>
+     *
+     * <p>
+     * If you have defined a shared access policy directly on the Event Hub itself, then copying the connection string
+     * from that Event Hub will result in a connection string that contains the path.
+     * </p>
+     *
+     * @param connectionString The connection string to use for connecting to the Event Hub instance. It is expected
+     *         that the Event Hub path and the shared access key properties are contained in this connection string.
+     * @return The updated EventHubClientBuilder object.
+     * @throws IllegalArgumentException if {@code connectionString} is null or empty. Or, the {@code connectionString}
+     *         does not contain the "EntityPath" key, which is the name of the Event Hub instance.
+     * @throws AzureException If the shared access signature token credential could not be created using the connection
+     *         string.
+     */
+    public EventHubClientBuilder credential(String connectionString) {
+        final ConnectionStringProperties properties = new ConnectionStringProperties(connectionString);
+        final TokenCredential tokenCredential;
+        try {
+            tokenCredential = new EventHubSharedAccessKeyCredential(properties.sharedAccessKeyName(),
+                properties.sharedAccessKey(), ClientConstants.TOKEN_VALIDITY);
+        } catch (InvalidKeyException | NoSuchAlgorithmException e) {
+            throw new AzureException("Could not create the EventHubSharedAccessKeyCredential.", e);
+        }
+
+        return credential(properties.endpoint().getHost(), properties.eventHubPath(), tokenCredential);
+    }
+
+    /**
+     * Sets the credential information given a connection string to the Event Hubs namespace and a path to a specific
+     * Event Hub instance.
+     *
+     * @param connectionString The connection string to use for connecting to the Event Hubs namespace; it is expected
+     *         that the shared access key properties are contained in this connection string, but not the Event Hub path.
+     * @param eventHubPath The path of the specific Event Hub to connect the client to.
+     * @return The updated EventHubClientBuilder object.
+     * @throws IllegalArgumentException if {@code connectionString} or {@code eventHubPath} is null or empty. Or, if the
+     *         {@code connectionString} contains the Event Hub path.
+     * @throws AzureException If the shared access signature token credential could not be created using the connection
+     *         string.
+     */
+    public EventHubClientBuilder credential(String connectionString, String eventHubPath) {
```

Comment:
This isn't a pattern we've hit yet - having fluent builders with multi-arg methods. Can you add this to your issue for post-preview 1 API to discuss?

Reply:
Created an issue to track this: https://github.com/Azure/azure-sdk-for-java/issues/4043